plugin/pkg/scheduler/factory/factory.go

```diff
@@ -1109,6 +1109,16 @@ func (factory *configFactory) MakeDefaultErrorFunc(backoff *util.PodBackoff, pod
 		} else {
 			if _, ok := err.(*core.FitError); ok {
 				glog.V(4).Infof("Unable to schedule %v %v: no fit: %v; waiting", pod.Namespace, pod.Name, err)
+			} else if errors.IsNotFound(err) {
+				if errStatus, ok := err.(errors.APIStatus); ok && errStatus.Status().Details.Kind == "node" {
+					// when node is not found, We do not remove the node right away. Trying again to get
+					// the node and if the node is still not found, then remove it from the scheduler cache.
+					_, err := factory.client.CoreV1().Nodes().Get(errStatus.Status().Details.Name, metav1.GetOptions{})
+					if err != nil && errors.IsNotFound(err) {
+						node := v1.Node{ObjectMeta: metav1.ObjectMeta{Name: errStatus.Status().Details.Name}}
+						factory.schedulerCache.RemoveNode(&node)
```

Comment:
Thanks, looks better. After removing the node, we also need to invalidate eCache predicates for the node. In order to do so, please add the following lines:
```
if factory.enableEquivalenceClassCache {
	factory.equivalencePodCache.InvalidateAllCachedPredicateItemOfNode(node.GetName())
}
```

Later on, we should refactor our code base so that the function that deletes a node from scheduler cache, also does the invalidation of the eCache.

Reply:
nit: it would be easier to read if you set
`nodeName := errStatus.Status().Details.Name`
and use `nodeName`.